weblate/vcs/mercurial.py

```diff
@@ -312,7 +312,7 @@ def describe(self):
     def push(self):
         """Push given branch to remote repository."""
         try:
-            self.execute(['push', '-r', '.'])
+            self.execute(['push'])
```

Comment:
Won't this fail in case there are more local branches? This can happen after switching branches. In Git, we always specify branch to push:

https://github.com/WeblateOrg/weblate/blob/8a52147265cbc593b71a02bb605def09e9025226/weblate/vcs/git.py#L370-L372

Reply:
On my end the `hg push -r .` won't work. Maybe I should verify the `hg push -b <branchname>` ?

Reply:
My knowledge of Mercurial is very limited, I'm just asking if that's correct. Looking at the docs using `-b` sounds reasonable.

Reply:
Looking into history, the `-r .` was added to handle current branch in 1157140b3a3.

Reply:
Going to test on my end the `self.execute(['push', '-b', self.branch])` ... :thinking:

Reply:
I did the test again, and odd... the "push -r ." works again :thinking:

Reply:
So, the "hg push -b <branch>" also works :fox_face: